components/cam/src/physics/rrtmg/radiation.F90

```diff
@@ -1823,7 +1833,11 @@ subroutine radiation_tend( state, ptend,pbuf, &
           ! stebol constant in mks units
           do i = 1,ncol
             tint(i,1) = state%t(i,1)
+#ifdef MAML
+            tint(i,pverp) = sqrt(sqrt(cam_in%lwup(i,ii)/stebol))
```

Comment:
If I follow this correctly, the loop is over `crm_nx_rad`, which in general is not the same as `crm_nx`. Have you considered what this means for the MAML implementation when they're not the same? Or does MAML require that `crm_nx_rad == crm_nx`?

Reply:
That is a great question. When I ran the MAML, I did not set up crm_nx_rad values, so the default is equal to crm_nx? The MAML needs all crm_nx columns have surface radiation flux passed to LND. So if crm_nx_rad is smaller than crm_nx, how can we decide the radiation flux for those (crm_nx - crm_nx_rad) columns? Do you have any suggestions?

Reply:
Jungmin and I had a long talk about this the other day. Ideally it would be great to be able to combine any number of CRM, land, and radiation columns. But just getting MAML to use less land instances than CRM columns is going to be hard enough, and is a much higher priority (not to mention getting the heterogeneity to work in both directions). I was hoping Guangxing and Jungmin could hash out a plan for these future developments while she is visiting PNNL. Right now I think it's ok to keep things as they are and do another small PR to add a check that enforces crm_nx_rad = crm_nx when MAML is used. I'm hoping the upgrades to MAML that would allow these other configurations can get implemented some time next year, either by Guangxing or Jungmin.

Reply:
Yeah, I don't have any particular suggestions for the overall solution. It seems like a design question that will benefit from thought and discussion. But if the requirement that `crm_nx_rad == crm_nx` is a current limitation to MAML, I would recommend adding in a check that calls `endrun` with a useful message if the user tries to both enable MAML and set `crm_nx_rad < crm_nx`.

Reply:
Yea, I guess it's worth putting in a fix for this now.

Reply:
I have put a check in crm_module.F90 as follows

#ifdef MAML
    if(crm_nx_rad.NE.crm_nx .or. crm_ny_rad.NE.crm_ny) then
      write(0,*) "crm_nx_rad and crm_ny_rad have to be equal to crm_nx and crm_ny in the MAML configuration"
      call endrun('crm main')
    end if
#endif

Reply:
Great, Can you make sure it's indented to match the rest of the code?